Bottom: 000ee20d0acf5984f9b875d4bc97dbbcfe1dc8d8
Top:    7aa2790ae36d6090e667a5b274cb372577abbba0
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-06 17:00:47 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 766ab67..4bf7069 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -236,6 +236,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
